src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -452,7 +453,7 @@ static void viewport_surface_smoothen_edge(
     uint8_t neighbourCorners[2] = { 0 };
     uint8_t ownCorners[2] = { 0 };
 
-    switch (edge)
+    switch (TEdge)
```

Comment:
Compilers are probably smart enough to see that only one here is used and inline that, but to guarantee that no branching takes place here, you could use `if constexpr` instead. Unfortunately, `switch constexpr` is not a thing (yet).